programs/test/dlopen.c

```diff
@@ -73,8 +74,16 @@ int main(void)
 #endif  /* MBEDTLS_X509_CRT_PARSE_C */
 
 #if defined(MBEDTLS_MD_C)
-    void *crypto_so = dlopen(CRYPTO_SO_FILENAME, RTLD_NOW);
-    CHECK_DLERROR("dlopen", CRYPTO_SO_FILENAME);
+    const char *crypto_so_filename = NULL;
+    void *crypto_so = dlopen(MBEDCRYPTO_SO_FILENAME, RTLD_NOW);
+    if (dlerror() == NULL) {
```

Comment:
Why is that required? This feels like it is fixing a programs's bug?

Reply:
It is required because the crypto library could be named libmbedcrypto or libtfpsacrypto depending the context the program is executed in. Thus we try both. Getting rid of libmbedcrypto comes with significant work thus I'd prefer to do it later.